File: MangaViewer/app/src/main/java/com/king/mangaviewer/util/GsonHelper.java
package com.king.mangaviewer.util;

import com.google.gson.Gson;

import java.lang.reflect.Type;

/**
 * Created by king on 2017-08-13.
 */

public class GsonHelper {
    private static Gson gson = new Gson();

    public static String toJson(Object obj){
        return gson.toJson(obj);
    }

    public static <T>T fromJson(String json, Type type){
        return gson.fromJson(json,type);
    }
}

File: MangaViewer/app/src/main/java/com/king/mangaviewer/util/GsonHelper.kt
package com.king.mangaviewer.util

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.king.mangaviewer.model.MangaUri

import java.lang.reflect.Type

/**
 * Created by king on 2017-08-13.
 */

object GsonHelper {
    val gson = Gson()

    fun toJson(obj: Any): String {
        return gson.toJson(obj)
    }

    fun <T> fromJson(json: String, type: Type): T {
        return gson.fromJson(json, type)
    }

}